src/spec-node/dockerCompose.ts

```diff
@@ -501,6 +509,19 @@ async function generateFeaturesComposeOverrideContent(
 	const userCommand = overrideCommand ? [] : composeCommand /* $ already escaped. */
 		|| (composeEntrypoint ? [/* Ignore image CMD per docker-compose.yml spec. */] : ((await imageDetails()).Config.Cmd || []).map(c => c.replace(/\$/g, '$$$$'))); // $ > $$ to escape docker-compose.yml's interpolation.
 
+	const hasGpuRequirement = config.hostRequirements?.gpu;
+	const addGpuCapability = hasGpuRequirement && await checkDockerSupportForGPU(params);
+	if (hasGpuRequirement && hasGpuRequirement !== 'optional' && !addGpuCapability) {
+		throw Error('Unable to detect GPU yet a GPU was required - consider marking it as "optional"');
```

Comment:
Could you log a warning instead? (Looks like you missed checking in this change.)

Reply:
Sorry, must have lost that when rebasing to the new main - I'll fix it now